web/libs/datamanager/src/components/Filters/types/List.jsx

```diff
@@ -14,9 +16,21 @@ export const VariantSelect = observer(({ filter, schema, onChange, multiple, val
   })();
 
   const FilterItem = filter.cellView?.FilterItem;
+
+  // Filter out users with firstName "Deleted" for annotator filters
+  const filteredItems =
+    filter.field?.alias === "annotators" || filter.field?.alias === "reviewers"
+      ? (items?.toJSON ? items.toJSON() : items).filter((item) => {
+          const user = sdk.store.users.find((u) => u.id === item);
+          return !(user?.firstName === "Deleted" && user?.lastName === "User");
+        })
+      : items?.toJSON
+        ? items.toJSON()
+        : items;
```

Comment:
Would be great to memoize this using `useMemo` will need to add `filter.field` `items` and `sdk.store.users` as dependencies.
This is beneficial since the filter through items will get worse with the length of that list and if we can reduce the number of times we do that, it would be great

Reply:
also Im guessing we do want to display deleted items else where - this is why we're not changing the request to only get non-deleted users from the backend?

Reply:
We have to still be able to pull the user record for purposes of displaying tasks annotators, reviewers, but searching for deleted users in the system is not feasible, since they are soft deleted and only kept for the purpose of not removing data produced by them in a given org's project. To be quite honest, I don't really see the need for soft deletion fwiw, we could have possibly just used a a simple method of marking user records in these selects with their role/status text like `First Last (Deactivated)` or `example@email.org (Inactive)` etc.

Filters with the ability to search the list plus better identification in the rendered list items would have been my pick for dealing with the cases of people that are not currently, or are no longer active in an org.